zephyr/lib/cpu.c

```diff
@@ -80,7 +85,13 @@ int cpu_enable_core(int id)
 		return 0;
 
 #if ZEPHYR_VERSION(3, 0, 99) <= ZEPHYR_VERSION_CODE
```

Comment:
why we need to check zephyr version? We won't ever go back to older zephyr releases.

Reply:
Are there plans to integrate some code from here into Zephyr itself as opposed to it staying in SOF? If so, then these checks are still worth it.

Reply:
@paulstelian97 yes, that the direction, IIUC there are still some dependencies to upstream.
I agree though, we dont need the version check as west manifest should allow us to specify supported Zephyr commit.
@tmleman does this PR need a west update ?

Reply:
> does this PR need a west update ?

@lgirdwood Yes, at first I need to merge zephyr PR and after that this can be merged with zephyr update. But regardless of that, everyone can review and approve.